### DISCUSSION TITLE
Use Tailwind directives within css in js

### DISCUSSION THREAD

**Discussion by deindesignpl** [ben-rogerson/twin.macro#423] [Ideas]
Idea would be to allow devs to use TW @apply within JSX. This would be helpful when using selectors like &::before. So instead of writing _${tw`text-black`}_ dev could write _@apply text-black_.

**Comment by ben-rogerson** [comment]
Hey, thanks for posting your suggestion.

Would you see the `@apply` function working any differently to `tw`?
If so, could you provide a sample of where you'd use it?

Or is it just a tw function that doesn't require backticks?

```js
css({
   span: {
     @apply text-black bg-white;
   }
})
```

^ That syntax is made to be written in css and wouldn't work in js outside template literals:

```js
styled.div`
  span {
    @apply text-black bg-white;  
  }
`
```
